hummingbird/ml/_container.py

```diff
@@ -40,139 +44,43 @@ def __init__(self, sparkml_model):
-                if device is not None and device.type != "cpu":
-                    inputs[i] = inputs[i].to(device)
-                variable_map[input_name] = inputs[i]
-
-            # Evaluate all the operators in the topology by properly wiring inputs \ outputs
-            for operator in self._operators:
-                pytorch_op = self._operator_map[operator.full_name]
-                pytorch_outputs = pytorch_op(*(variable_map[input] for input in operator.input_full_names))
-
-                if len(operator.output_full_names) == 1:
-                    variable_map[operator.output_full_names[0]] = pytorch_outputs
-                else:
-                    for i, output in enumerate(operator.output_full_names):
-                        variable_map[output] = pytorch_outputs[i]
-
-            # Prepare and return the output.
-            if len(self._output_names) == 1:
-                return variable_map[self._output_names[0]]
-            else:
-                return list(variable_map[output_name] for output_name in self._output_names)
-
-
-class PyTorchTorchscriptSklearnContainer(ABC):
-    """
-    Base container for PyTorch and TorchScript models.
-    The container allows to mirror the Sklearn API.
-    """
+        Base container abstract class allowing to mirror the Sklearn API.
+        Thanks to *SklearnContainer* we can use the `predict`, `predict_proba` etc. API of Sklearn
+        also over the models generated by Hummingbird (irrespective of the selected backend).
 
-    def __init__(self, model, extra_config={}):
-        """
         Args:
-            model: A pytorch or torchscript model
-            extra_config: Some additional configuration parameter
+            model: A model
+            n_threads: How many threads should be used by the containter to run the model. None means use all threads.
+            batch_size: If different than None means split the input into batch_size partitions and score one partition at a time.
+            extra_config: Some additional configuration parameter.
         """
         self._model = model
+        self._n_threads = n_threads
+        self._batch_size = batch_size
         self._extra_config = extra_config
 
     @property
     def model(self):
         return self._model
 
-    def to(self, device):
-        """
-        Set the target device for the model.
 
-        Args:
-            device: The target device.
-        """
-        self.model.to(device)
+class PyTorchTorchscriptSklearnContainer(SklearnContainer):
+    """
+    Base container for PyTorch and TorchScript models.
+    """
+
+    def __init__(self, model, n_threads=None, batch_size=None, extra_config={}):
+        super(PyTorchTorchscriptSklearnContainer, self).__init__(model, n_threads, batch_size, extra_config)
+
+        assert self._n_threads is not None
+
+        if torch.get_num_interop_threads() != 1:
```

Comment:
Can you add a comment here on why we must first reset the number of threads to 1?  I know it's the constructor, but at first glance it's confusing.